src/library/client.lua

```diff
@@ -33,81 +115,45 @@ function client.get_raw_finale_version(major, minor, build)
 end
 
 --[[
-% supports_smufl_fonts()
+% supports
 
-Returns true if the current client supports SMuFL fonts.
-
-: (boolean)
-]]
-function client.supports_smufl_fonts()
-    return finenv.RawFinaleVersion >= client.get_raw_finale_version(27, 1)
-end
+Checks the client supports a given feature. Returns true if the client
+supports the feature, false otherwise.
 
---[[
-% supports_category_save_with_new_type()
+To assert the client must support a feature, use `client.assert_supports`.
 
-Returns true if the current client supports FCCategory::SaveWithNewType().
+For a list of valid features, see the [`features` table in the codebase](https://github.com/finale-lua/lua-scripts/blob/master/src/library/client.lua#L52).
 
+@ feature (string) The feature the client should support.
 : (boolean)
 ]]
-function client.supports_category_save_with_new_type()
-    return finenv.StringVersion >= "0.58"
+function client.supports(feature)
+    return features[feature].test()
 end
 
 --[[
-% supports_finenv_query_invoked_modifier_keys()
+% assert_supports
 
-Returns true if the current client supports finenv.QueryInvokedModifierKeys().
+Asserts that the client supports a given feature. If the client doesn't
+support the feature, this function will throw an friendly error then
+exit the program.
 
-: (boolean)
-]]
-function client.supports_finenv_query_invoked_modifier_keys()
-    return finenv.IsRGPLua and finenv.QueryInvokedModifierKeys
-end
+To simply check if a client supports a feature, use `client.supports`.
 
---[[
-% supports_retained_state()
-
-Returns true if the current client supports retaining state between runs.
-
-: (boolean)
-]]
-function client.supports_retained_state()
-    return finenv.IsRGPLua and finenv.RetainLuaState ~= nil
-end
-
---[[
-% supports_modeless_dialog()
-
-Returns true if the current client supports modeless dialogs.
+For a list of valid features, see the [`features` table in the codebase](https://github.com/finale-lua/lua-scripts/blob/master/src/library/client.lua#L52).
 
+@ feature (string) The feature the client should support.
 : (boolean)
 ]]
-function client.supports_modeless_dialog()
-    return finenv.IsRGPLua
-end
-
---[[
-% supports_clef_changes()
-
-Returns true if the current client supports changing clefs.
-
-: (boolean)
-]]
-function client.supports_clef_changes()
-    return finenv.IsRGPLua or finenv.StringVersion >= "0.60"
-end
-
---[[
-% supports_custom_key_signatures()
-
-Returns true if the current client supports changing clefs.
-
-: (boolean)
-]]
-function client.supports_custom_key_signatures()
-    local key = finale.FCKeySignature()
-    return finenv.IsRGPLua and key.CalcTotalChromaticSteps
+function client.assert_supports(feature)
+    if not client.supports(feature) then
+        if features[feature].error then
+            error(features[feature].error, 0)
```

Comment:
What about having level `2` if in debug mode and level `0` otherwise? If I was writing a plugin and I got this error, I'd want to know exactly where it was coming from.

Reply:
That makes sense. Do you happen to know how to determine if you're running in debug mode?

Reply:
`finenv.DebugEnabled and 2 or 0`